scripts/skills/actives/rf_encourage_skill.nut

```diff
@@ -44,64 +46,68 @@ this.rf_encourage_skill <- this.inherit("scripts/skills/skill", {
 	}
 
 	function onVerifyTarget( _originTile, _targetTile )
 	{
 		if (!this.skill.onVerifyTarget(_originTile, _targetTile))
-		{
 			return false;
-		}
 
 		local target = _targetTile.getEntity();
+		if (target.getCurrentProperties().IsStunned || target.getMoraleState() == ::Const.MoraleState.Fleeing || target.getMoraleState() >= ::Const.MoraleState.Confident || target.getMoraleState() >= target.m.MaxMoraleState)
+			return false;
+
 		local actor = this.getContainer().getActor();
-		return actor.getFaction() == target.getFaction() && target.getMoraleState() < actor.getMoraleState() && this.__canBeEncouraged(target);
+		return actor.getFaction() == target.getFaction() && actor.getMoraleState() - target.getMoraleState() >= actor.getTile().getDistanceTo(target.getTile());
 	}
 
-	function onUse( _user, _targetTile )
+	function isUsable()
 	{
-		local target = _targetTile.getEntity();
-
-		target.checkMorale(1, this.getEncourageBonus(), ::Const.MoraleCheckType.Default);
-		this.__markEntityWithRound(target);
-
-		return true;
+		return !this.m.IsSpent && this.skill.isUsable();
 	}
 
-// New Functions
-	function getEncourageBonus()
+	function onUse( _user, _targetTile )
 	{
-		local encourageBonus = this.getContainer().getActor().getCurrentProperties().getBravery() * this.m.EncourageBonusFraction;
-		return ::Math.max(0, encourageBonus);	// The bonus can never be negative
+		_targetTile.getEntity().checkMorale(1, this.getEncourageBonus(), ::Const.MoraleCheckType.Default);
+		this.m.IsSpent = true;
+		return true;
 	}
 
-	function __canBeEncouraged( _targetEntity )
+	function onTurnStart()
 	{
-		// The MaxMoraleState part might be confusing for the player because it's not mentioned in the tooltip but it's just logical
-		if (_targetEntity.getCurrentProperties().IsStunned || _targetEntity.getMoraleState() >= _targetEntity.m.MaxMoraleState)
-			return false;
-
-		return this.__getRoundOfMarkedEntity(_targetEntity) < ::Time.getRound();
+		this.m.IsSpent = false;
 	}
 
-	// Write the current round number in a tactical manager flag that consists of the id of an entity and the id of this skill
-	function __markEntityWithRound( _targetEntity )
+	function onCombatFinished()
 	{
-		::Tactical.Entities.getFlags().set(_targetEntity.getID() + this.getID() + "MarkedWithRoundNumber", ::Time.getRound());
+		this.skill.onCombatFinished();
+		this.m.IsSpent = false;
 	}
 
-	// Return the round number imprinted into a tactical manager flag for that entity; Returns 0 if no flag exists yet for it
-	function __getRoundOfMarkedEntity( _targetEntity )
+// New Functions
+	function getEncourageBonus()
 	{
-		if (::Tactical.Entities.getFlags().has(_targetEntity.getID() + this.getID() + "MarkedWithRoundNumber"))
-		{
-			return ::Tactical.Entities.getFlags().get(_targetEntity.getID() + this.getID() + "MarkedWithRoundNumber");
-		}
-
-		return 0;
+		local encourageBonus = this.getContainer().getActor().getCurrentProperties().getBravery() * this.m.EncourageBonusFraction;
```

Comment:
Vanilla floors this value in the `rally_the_troops` file. We should also do that or round it at least so that the checkMorale function doesn't receive an unexpected value.

Though I don't understand why we not already see decimal places on the value in the current implementation

Reply:
We are using `::Math.max` function here already and that floors it.